gips/0054-timeline-aggregation-protocol.md

```diff
@@ -0,0 +1,270 @@
+
+Please note those checks are done as part of the critical path when an Indexer is serving a query.
+This might impact the total serving time.
+Hence TAP enables the Indexers to postpone all/or some of those checks to be executed during the RAV request preparation,
+as presented in fig. 5.
+In short, during RAV request preparation TAP manager checks if a given receipt passed all checks - and if some are missing it will run them at the very moment.
+
+| ![tap_manager_rav_request_checks.png](../assets/gip-0054/tap_manager_rav_request_checks.png) |
+| - |
+| <b>Fig. 5: (Mandatory) receipt checks performed by the Indexer when preparing RAV request</b>|
+
+Please refer to [TAP Manager documentation](https://docs.rs/tap_core/0.1.0/tap_core/tap_manager/struct.Manager.html) for more details on this topic.
+
+
+
+### Gateway Receipt Aggregator
+
+Gateway Receipt Aggregator is a A JSON-RPC service that lets clients request an aggregate receipt (RAV) based on a list of individual receipts. High-level specification can be found [TAP Aggregator README](https://docs.rs/crate/tap_aggregator/0.1.0).
+
+Detailed documentation can be found in [TAP Aggregator docs.rs](https://docs.rs/tap_core/0.1.0/tap_core/).
+
+## Smart Contracts (Tom/Bryan | Status: OK)
+
+TAPVerifier - A contract for verifying receipt aggregation vouchers.
+
+Collateral - This contract allows `senders` to deposit collateral for specific `receivers`, which can later be redeemed using Receipt Aggregate Vouchers (`RAV`) signed by an authorized `signer`. `Senders` can deposit collateral for `receivers`, authorize `signers` to create signed `RAVs`, and withdraw collateral after a set `thawingPeriod` number of seconds. `Receivers` can redeem signed `RAVs` to claim collateral.
+This contract uses the `TAPVerifier` contract for recovering signer addresses from `RAVs`.
+
+AllocationIDTracker - This contract tracks the allocation IDs of the RAVs that have been submitted to ensure that each allocation ID is only used once. It is external to the collateral contract to allow for updating the collateral contract without losing the list of used allocation IDs.
+This contract is intended to be used with the `Collateral` contract.
+
+Implementation along with detailed API documentation of TAP contracts can be found [here](https://github.com/semiotic-ai/timeline-aggregation-protocol-contracts).
+
+# Backwards Compatibility
+
+## Gateway:
+Indexers are currently expected to report their software version (via the `/version` path of indexer-service). During the transition period, the gateway will use this version to determine if the indexer can be expected to support TAP. If so, the gateway will only send receipts to that indexer using TAP. Otherwise, the current Scalar receipt format will be used. This will ensure that any query sent to an indexer from a gateway will only be associated with one receipt, and that receipt will only be associated with one of the two Scalar protocol versions.
```

Comment:
The Gateway is taking the burden of handling the upgrade - is it worth specifying the length of the transition period / providing some kind of "push" to encourage indexers to upgrade?

Reply:
Yes, we will have to plan this and create some kind of a campaign, motivating the indexers to update their stack once everything will be deployed. TBD.